src/ucs/sys/sock.c

```diff
@@ -100,22 +100,78 @@ ucs_status_t ucs_netif_ioctl(const char *if_name, unsigned long request,
     return status;
 }
 
-int ucs_netif_is_active(const char *if_name)
+int ucs_netif_get_addr(const char *if_name, uint8_t sa_family,
+                       struct sockaddr *saddr, struct sockaddr *netmask)
 {
-    ucs_status_t status;
-    struct ifreq ifr;
+    ucs_status_t status = UCS_ERR_NO_DEVICE;
+    struct ifaddrs *ifa;
+    struct ifaddrs *ifaddrs;
+    const struct sockaddr_in6 *saddr6;
+    size_t addrlen;
 
-    status = ucs_netif_ioctl(if_name, SIOCGIFADDR, &ifr);
-    if (status != UCS_OK) {
-        return 0;
+    if (getifaddrs(&ifaddrs)) {
```

Comment:
we removed the call to getifaddrs because it caused errors/stalls during init
pls use other method to get netif address, like reading /proc/net/if_net6
also, maybe cache the result (in global scope) to avoid lookup every time?

Reply:
it seems we are already using getifaddrs in the codebase (https://github.com/openucx/ucx/blob/master/src/ucs/sys/sock.c#L896) and also gtests. do you have more details on the issue?

Reply:
sorry, the issue was with `if_nameindex` (see https://github.com/openucx/ucx/pull/1276/commits/9b2e4aee95e3848211009b0f9b81d383bcd68d96)
so it's ok to use getifaddrs